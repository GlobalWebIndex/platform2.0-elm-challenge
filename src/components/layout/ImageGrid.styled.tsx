import styled from '@emotion/styled';
import { Grid } from '@mui/material';

const StyledImageGrid = styled(Grid)({
    display: 'grid',
    justifyContent: 'center',
    alignItems: 'center',
    gridTemplateColumns: 'repeat(auto-fill, 200px)',
    gap: '20px',
    padding: '20px 40px',
});

const ImageGrid = () => {
    return <StyledImageGrid></StyledImageGrid>;
};

export default ImageGrid;
