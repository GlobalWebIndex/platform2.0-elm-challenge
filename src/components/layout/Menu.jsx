import React from "react"
import { useMenuStyles } from "./Menu.styles"

function Menu() {
  const classes = useMenuStyles()

  return (
    <ul className={classes.menu}>
      <li className={classes.menuItem}>
        <a href="/" className={classes.menuItemLink}>
          Images
        </a>
      </li>

      <li className={classes.menuItem}>
        <a href="/breeds" className={classes.menuItemLink}>
          Breeds
        </a>
      </li>

      <li className={classes.menuItem}>
        <a href="/favourites" className={classes.menuItemLink}>
          Favourites
        </a>
      </li>
    </ul>
  )
}

export default Menu
