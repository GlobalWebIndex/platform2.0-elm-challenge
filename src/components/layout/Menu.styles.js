import { createUseStyles } from "react-jss"

export const useMenuStyles = createUseStyles({
  menu: {
    margin: 0,
    padding: 0,
    listStyle: "none",
    display: "flex",
    gridTemplateColumns: "repeat(6, 1fr)",
    position: "relative",
  },
  menuItem: {
    textAlign: "right",
    margin: [0, 20, 0, 0],
    padding: 0,
    display: "inline-flex",
  },
  menuItemLink: {
    font: "700 15px/20px Helvetica",
    textDecoration: "none",
    color: "#000",
    display: "block",
    width: "100%",
    letterSpacing: "0.5px",

    "&:hover": {
      color: "#e2b71c",
    },
  },
})
