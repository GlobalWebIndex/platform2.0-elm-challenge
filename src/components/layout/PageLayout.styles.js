import { createUseStyles } from "react-jss"
import { breakpoints, mediaBreakpointUp } from "../../util"

export const usePageLayoutStyles = createUseStyles({
  pageLayout: {
    display: "grid",
  },
  menuWrapper: {
    display: "grid",
    position: "relative",

    [mediaBreakpointUp(breakpoints.md)]: {
      justifySelf: "end",
    },
  },
  contentWrapper: {},
})
