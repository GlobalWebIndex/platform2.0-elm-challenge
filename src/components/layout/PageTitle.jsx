import React from "react"
import { usePageTitleStyles } from "./PageTitle.styles"

function PageTitle({ title }) {
  const classes = usePageTitleStyles()

  return (
    <h1 className={classes.pageTitle} data-testid="page-title">
      {title}
    </h1>
  )
}

export default PageTitle
