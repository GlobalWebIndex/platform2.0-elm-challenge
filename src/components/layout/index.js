export { default as Menu } from "./Menu"
export { default as PageLayout } from "./PageLayout"
export { default as PageTitle } from "./PageTitle"