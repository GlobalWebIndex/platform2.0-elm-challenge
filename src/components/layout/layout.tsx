import React from 'react';
import {Outlet} from 'react-router';
import {Link} from 'react-router-dom';
import styles from './styles.scss';

export const Layout = (): JSX.Element => (
  <>
    <div className={styles.navigation}>
      <h1>Cat lover</h1>
      <nav>
        <ul>
          <li>
            <Link to="cats">Cats</Link>
          </li>
          <li>
            <Link to="breeds">Breeds</Link>
          </li>
        </ul>
      </nav>
    </div>
    <Outlet />
  </>
);
