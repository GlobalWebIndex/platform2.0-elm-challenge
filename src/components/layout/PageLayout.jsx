import React from "react"
import Menu from "./Menu"
import { usePageLayoutStyles } from "./PageLayout.styles"

function PageLayout({ children }) {
  const classes = usePageLayoutStyles()

  return (
    <div className={classes.pageLayout}>
      <div className={classes.menuWrapper}>
        <Menu />
      </div>
      <div className={classes.contentWrapper}>{children}</div>
    </div>
  )
}

export default PageLayout
