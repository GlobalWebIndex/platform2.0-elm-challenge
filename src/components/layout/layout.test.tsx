import React from 'react';
import {MemoryRouter} from 'react-router-dom';
import {render} from '@testing-library/react';
import {Layout} from './layout';

describe('<Layout />', () => {
  it('should render navigation links', () => {
    const {container} = render(<Layout />, {wrapper: MemoryRouter});

    expect(container).toMatchSnapshot();
  });
});
