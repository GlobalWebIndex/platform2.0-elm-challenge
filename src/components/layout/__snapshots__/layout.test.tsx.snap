// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> should render navigation links 1`] = `
<div>
  <div
    class="navigation"
  >
    <h1>
      Cat lover
    </h1>
    <nav>
      <ul>
        <li>
          <a
            href="/cats"
          >
            Cats
          </a>
        </li>
        <li>
          <a
            href="/breeds"
          >
            Breeds
          </a>
        </li>
      </ul>
    </nav>
  </div>
</div>
`;
